shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -187,91 +234,134 @@ public Intent build(@NonNull Context context) {
+   *   of "io.flutter.embedding.android.NormalTheme" and set the resource to your normal
+   *   theme, e.g., {@code android:resource="@style/MyNormalTheme}.
+   * </ol>
+   * With the above settings, your launch theme will be used when loading the app, and
+   * then the theme will be switched to your normal theme once the app has initialized.
+   * <p>
+   * Do not change aspects of system chrome between a launch theme and normal theme. Either define
+   * both themes to be fullscreen or not, and define both themes to display the same status bar and
+   * navigation bar settings. If you wish to adjust system chrome once your Flutter app renders, use
+   * platform channels to instruct Android to do so at the appropriate time. This will avoid any
+   * jarring visual changes during app startup.
    */
-  private void configureWindowForTransparency() {
-    BackgroundMode backgroundMode = getBackgroundMode();
-    if (backgroundMode == BackgroundMode.transparent) {
-      getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
-      getWindow().setFlags(
-        WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
-        WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
-      );
+  private void switchLaunchThemeForNormalTheme() {
+    try {
+      ActivityInfo activityInfo = getPackageManager().getActivityInfo(getComponentName(), PackageManager.GET_META_DATA);
+      if (activityInfo.metaData != null) {
+        int normalThemeRID = activityInfo.metaData.getInt(NORMAL_THEME_META_DATA_KEY, -1);
+        if (normalThemeRID != -1) {
+          setTheme(normalThemeRID);
+        }
+      } else {
+        Log.d(TAG, "Using the launch theme as normal theme.");
+      }
+    } catch (PackageManager.NameNotFoundException exception) {
+      Log.e(TAG, "Could not read meta-data for FlutterActivity. Using the launch theme as normal theme.");
     }
   }
 
   /**
-   * Cover all visible {@code Activity} area with a {@code View} that paints everything the same
-   * color as the {@code Window}.
+   * Extracts a {@link Drawable} from the {@code Activity}'s {@code windowBackground}.
    * <p>
-   * This cover {@code View} should be displayed at the very beginning of the {@code Activity}'s
-   * lifespan and then hidden once Flutter renders its first frame. The purpose of this cover is to
-   * cover {@link FlutterSurfaceView}, which briefly displays a black rectangle before it can make
-   * itself transparent.
+   * Returns null if no {@code windowBackground} is set for the activity.
    */
-  private void showCoverView() {
-    if (getBackgroundMode() == BackgroundMode.transparent) {
-      // Don't display an opaque cover view if the Activity is intended to be transparent.
-      return;
+  @SuppressWarnings("deprecation")
+  private Drawable getLaunchScreenDrawableFromActivityTheme() {
+    TypedValue typedValue = new TypedValue();
+    if (!getTheme().resolveAttribute(
+        android.R.attr.windowBackground,
```

Comment:
I think the splash guide that is up for PR should be sufficient, no? Also, I don't believe running a Flutter upgrade would ever change these files, would it?

Reply:
So I made a flutter app in the past, it had a project template where the android manifest makes an activity with a windowBackground. They flutter upgrade and that activity which was a io.flutter.app.FlutterActivity is now a shim to FullFlutterActivity. The FullFlutterActivity would still show that windowBackground and respond to the first frame callback and remove the windowBackground the same way right?

Reply:
I'm still curious how such an upgrade occurs. Are you talking about a situation with a .android directory? Because my understanding of a normal project is that we don't touch the project files...do I have that wrong?

Reply:
I think we talked offline